Formula/d/decasify.rb

```diff
@@ -28,11 +29,22 @@ class Decasify < Formula
 
   uses_from_macos "zlib"
 
+  # shell completion file permission patch, upstream pr ref, https://github.com/alerque/decasify/pull/37
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/280b19b530f7328464e7aacd31a3faa621aa137f/decasify/0.8.0-fix-shell-permissions.patch"
+    sha256 "0fd8046981413341698166b1942020956dfa5ee6a4874ee418557179e61a8a6f"
+  end
+
   def install
     system "./bootstrap.sh" if build.head?
     system "./configure", "--disable-silent-rules", *std_configure_args
     system "make"
     system "make", "install"
+
+    bash_completion.install "completions/decasify"
+    fish_completion.install "completions/decasify.fish"
+    zsh_completion.install "completions/_decasify"
+    man1.install "decasify.1"
```

Comment:
Why is this necessary? This project is setup to install all of these using `make install` if it was able to detect the right place to put them during configure. If not there are configure options to set the target paths, after which the install phase should install them properly. See `--with-$SHELL-cempletion-dir=PATH` for bash/fish/zsh. These should have had reasonable defaults to start with including leveraging info reported by `pkg-config` about where the shell data dirs are.

The man page likewise should automatically be installed to a normal location.

If there is a mistake upstream I'd like to get it fixed because this should have worked without any extra manual packaging/installation steps needed.

Reply:
I did not explore too deep with the args, feel free to shoot another PR on it. (so far, I just saw the file permission issues)

also, using the brew commands to do the install is pretty common for the rust project I see.

Reply:
actually you are right, no need to do the manual installation, I will revert the change now

```
==> make install
/Applications/Xcode.app/Contents/Developer/usr/bin/make  install-am
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/share/licenses/decasify'
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/bin'
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/share/doc/decasify'
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/share/bash-completion/completions'
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/share/fish/vendor_completions.d'
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/share/man/man1'
 ./build-aux/install-sh -c -d '/opt/homebrew/Cellar/decasify/0.8.0/share/zsh/site-functions'
 /usr/bin/install -c -m 644 completions/decasify.fish '/opt/homebrew/Cellar/decasify/0.8.0/share/fish/vendor_completions.d'
 /usr/bin/install -c -m 644 LICENSE.md '/opt/homebrew/Cellar/decasify/0.8.0/share/licenses/decasify'
 /usr/bin/install -c -m 644 completions/decasify '/opt/homebrew/Cellar/decasify/0.8.0/share/bash-completion/completions'
 /usr/bin/install -c -m 644 README.md CHANGELOG.md '/opt/homebrew/Cellar/decasify/0.8.0/share/doc/decasify'
 /usr/bin/install -c -m 644 completions/_decasify '/opt/homebrew/Cellar/decasify/0.8.0/share/zsh/site-functions'
  /usr/bin/install -c decasify '/opt/homebrew/Cellar/decasify/0.8.0/bin'
 /usr/bin/install -c -m 644 decasify.1 '/opt/homebrew/Cellar/decasify/0.8.0/share/man/man1'
==> Cleaning
==> Fixing /opt/homebrew/Cellar/decasify/0.8.0/bin/decasify permissions from 755 to 555
==> Finishing up
/opt/homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::NullLoader): loading decasify
ln -s ../Cellar/decasify/0.8.0/bin/decasify decasify
ln -s ../../../Cellar/decasify/0.8.0/share/bash-completion/completions/decasify decasify
ln -s ../../Cellar/decasify/0.8.0/share/doc/decasify decasify
ln -s ../../../Cellar/decasify/0.8.0/share/fish/vendor_completions.d/decasify.fish decasify.fish
ln -s ../../Cellar/decasify/0.8.0/share/licenses/decasify decasify
ln -s ../../../Cellar/decasify/0.8.0/share/man/man1/decasify.1 decasify.1
ln -s ../../../Cellar/decasify/0.8.0/share/zsh/site-functions/_decasify _decasify
==> Summary
🍺  /opt/homebrew/Cellar/decasify/0.8.0: 14 files, 2.2MB, built in 39 seconds
```

```
$ tree /opt/homebrew/Cellar/decasify/0.8.0
/opt/homebrew/Cellar/decasify/0.8.0
├── CHANGELOG.md
├── INSTALL_RECEIPT.json
├── LICENSE.md
├── README.md
├── bin
│   └── decasify
├── sbom.spdx.json
└── share
    ├── bash-completion
    │   └── completions
    │       └── decasify
    ├── doc
    │   └── decasify
    │       ├── CHANGELOG.md
    │       └── README.md
    ├── fish
    │   └── vendor_completions.d
    │       └── decasify.fish
    ├── licenses
    │   └── decasify
    │       └── LICENSE.md
    ├── man
    │   └── man1
    │       └── decasify.1
    └── zsh
        └── site-functions
            └── _decasify

15 directories, 13 files
```

Reply:
@alerque thanks for the input in here, ❤️ 

gonna revert in https://github.com/Homebrew/homebrew-core/pull/202363

Reply:
> also, using the brew commands to do the install is pretty common for the rust project I see

Sadly `cargo` explicitly caters *only* to direct end user usage (i.e. to user space such as `~/.cargo/bin/`) not packaging or system installation. In fact it is actively hostile to integration with packaging or full blown build systems. This isn't an uncommon stance for language ecosystem specific tooling (cargo, pip, npm, etc. also suffer a bit) but cargo is particularly bad (e.g. Cargo has denied repeated attempts to make the final build artifact directory easy to query or derive, `clap` is moving *away* from making completions packagable at all, etc.).

That being said many/most Rust based projects are lazy and go with the flow ... and don't actually include any tooling that is useful for packaging or installation. Some slightly more adventurous ones may have an `xtask` script or two to deal with checking for a system dependency or shuffling around some assets. Relatively few Rust projects integrate with a full projet build system (autotools, meson, cmake, bazel, etc.). Of the ones that do, autotools is probably one of the *least* common to see it integrated with, but that's what this project does.

As such it should basically be treated like any other autotools based project that might be in C or some other language, and ignore the fact that there is Rust stuff involved. I've worked hard (not just for this project but many others I use these macros for) to both allow `cargo install` without any trappings for quick user space installs while also integrating a full build system in the project that handles completions, man pages, etc.